x-pack/legacy/plugins/maps/common/descriptor_types.d.ts

```diff
@@ -18,3 +18,30 @@ export interface ILayerDescriptor {
   id: string;
   label?: string;
 }
+
+interface IFieldMetaDescriptor {
+  [key: string]: string | number | unknown;
+}
+
+interface IRangeFieldMetaDescriptor extends IFieldMetaDescriptor {
+  min: number;
+  max: number;
+  delta: number;
+}
+
+interface ICategoryFieldMetaDescriptor extends IFieldMetaDescriptor {
```

Comment:
`ICategoryFieldMetaDescriptor ` is not a descriptive enough name. There are two objects that could be called `ICategoryFieldMetaDescriptor `.
1) fieldMetaFromFeatures
2) fieldMeta from elasticsearch terms aggregation call.

This instance is the first, field meta from features. How about renaming to `ICategoryFieldMetaDescriptor`. I think you can drop the `Descriptor` part here since there is not fieldMeta object that this could be confused with.

Then, how about adding a comment on where these values come from.